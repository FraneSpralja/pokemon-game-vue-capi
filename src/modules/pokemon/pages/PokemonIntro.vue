<template>
    <section class="flex flex-col justify-center items-center w-screen h-screen">
        <div class="intro-form">
            <div class="form__box">
                <label for="name">Entrenador</label>
                <input type="text" name="name" v-model="name">
            </div>
            <div class="form__box">
                <label for="name" class="block">Dificultad</label>
                <div class="form__box--container flex">
                    <label v-for="dif in difficults" :key="dif"> {{ dif }} pokemons
                        <input type="radio" :value="dif" v-model="pokemonLevel">
                    </label>
                </div>
            </div>
            <div class="form__box">
                <button class="bg-blue-500 text-white p-2 rounded-md w-fit mx-auto hover:bg-blue-800"
                    @click="emit('setupGame', { name, level: pokemonLevel })">
                    Iniciar Batalla
                </button>
            </div>
        </div>
    </section>
</template>

<script setup lang="ts">
import { reactive, ref } from 'vue';

const emit = defineEmits<{
    setupGame: [{ name: string, level: number }]
}>();

const difficults = reactive([3, 4, 5, 7])
const name = ref<string>("Ash")
const pokemonLevel = ref<number>(4)

</script>

<style scoped>
.form__box {
    @apply flex flex-col mb-4;
}

.form__box--container {
    @apply flex gap-4;

    label {
        @apply flex justify-start items-center gap-2;
    }
}
</style>