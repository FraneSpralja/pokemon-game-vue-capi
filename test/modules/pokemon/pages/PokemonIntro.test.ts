import PokemonIntro from '@/modules/pokemon/pages/PokemonIntro.vue';
import { mount } from '@vue/test-utils';
import { describe, expect, test } from 'vitest';

describe('<PokemonIntro />', () => {

    test('should render with default values', () => {

        const wrapper = mount(PokemonIntro)

        const inputs = wrapper.findAll('input[type="radio"]')
        expect(wrapper.html()).toMatchSnapshot();
        expect(inputs.length).toBe(4)
    })

    test('should correctly emit setupGame', async() => {

        const name = 'Gary'

        const wrapper = mount(PokemonIntro)

        const input = wrapper.find('input[type="text"]')
        await input.setValue(name)

        const radioInput = wrapper.find('input[type="radio"][value="4"]')
        await radioInput.trigger('change')

        const button = wrapper.find('button')
        await button.trigger('click')

        expect(wrapper.emitted('setupGame')?.[0]).toEqual([{
            name: name,
            level: expect.any(Number)
        }])
    })
})