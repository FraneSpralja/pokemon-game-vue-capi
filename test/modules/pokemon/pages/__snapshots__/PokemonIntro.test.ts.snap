// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PokemonIntro /> > should render with default values 1`] = `
"<section data-v-e2f8eb3c="" class="flex flex-col justify-center items-center w-screen h-screen">
  <div data-v-e2f8eb3c="" class="intro-form">
    <div data-v-e2f8eb3c="" class="form__box"><label data-v-e2f8eb3c="" for="name">Entrenador</label><input data-v-e2f8eb3c="" type="text" name="name"></div>
    <div data-v-e2f8eb3c="" class="form__box"><label data-v-e2f8eb3c="" for="name" class="block">Dificultad</label>
      <div data-v-e2f8eb3c="" class="form__box--container flex"><label data-v-e2f8eb3c="">3 pokemons <input data-v-e2f8eb3c="" type="radio" value="3"></label><label data-v-e2f8eb3c="">4 pokemons <input data-v-e2f8eb3c="" type="radio" value="4"></label><label data-v-e2f8eb3c="">5 pokemons <input data-v-e2f8eb3c="" type="radio" value="5"></label><label data-v-e2f8eb3c="">7 pokemons <input data-v-e2f8eb3c="" type="radio" value="7"></label></div>
    </div>
    <div data-v-e2f8eb3c="" class="form__box"><button data-v-e2f8eb3c="" class="bg-blue-500 text-white p-2 rounded-md w-fit mx-auto hover:bg-blue-800"> Iniciar Batalla </button></div>
  </div>
</section>"
`;
